src/dotnet/Azure.ClientSdk.Analyzers/Azure.ClientSdk.Analyzers/ModelFactoryAnalyzer.cs

```diff
@@ -0,0 +1,228 @@
+                    if (outputModel != null)
+                    {
+                        outputModels.Add(outputModel);
+                    }
+                }
+            }
+        }
+
+        private static void ExtractReturnTypesFromModelFactory(INamedTypeSymbol factoryType, HashSet<ITypeSymbol> modelFactoryMethods)
+        {
+            foreach (var member in factoryType.GetMembers())
+            {
+                if (member is IMethodSymbol method && 
+                    method.DeclaredAccessibility == Accessibility.Public && 
+                    method.IsStatic)
+                {
+                    // Add the return type of the factory method
+                    modelFactoryMethods.Add(method.ReturnType);
+                }
+            }
+        }
+
+        private static ITypeSymbol ExtractOutputModelFromReturnType(ITypeSymbol returnType)
+        {
+            ITypeSymbol unwrappedType = returnType;
+
+            // Unwrap Task<T>
+            if (returnType is INamedTypeSymbol namedType && 
+                namedType.IsGenericType && 
+                namedType.Name == TaskTypeName)
+            {
+                unwrappedType = namedType.TypeArguments.FirstOrDefault();
+                if (unwrappedType == null) return null;
+            }
+
+            ITypeSymbol modelType = null;
+
+            // Check for Azure client method return types and extract the model type
+            if (IsOrImplements(unwrappedType, ResponseTypeName, AzureNamespace) ||
+                IsOrImplements(unwrappedType, NullableResponseTypeName, AzureNamespace))
+            {
+                if (unwrappedType is INamedTypeSymbol responseType && responseType.IsGenericType)
+                {
+                    modelType = responseType.TypeArguments.FirstOrDefault();
+                }
+            }
+            else if (IsOrImplements(unwrappedType, OperationTypeName, AzureNamespace))
+            {
+                if (unwrappedType is INamedTypeSymbol operationType && operationType.IsGenericType)
+                {
+                    modelType = operationType.TypeArguments.FirstOrDefault();
+                }
+            }
+            else if (IsOrImplements(returnType, PageableTypeName, AzureNamespace) ||
+                     IsOrImplements(returnType, AsyncPageableTypeName, AzureNamespace))
+            {
+                if (returnType is INamedTypeSymbol pageableType && pageableType.IsGenericType)
+                {
+                    modelType = pageableType.TypeArguments.FirstOrDefault();
+                }
+            }
+
+            // Only return user-defined types, not built-in types
+            if (modelType != null && IsUserDefinedModelType(modelType))
+            {
+                return modelType;
+            }
+
+            return null;
+        }
+
+        private static bool IsUserDefinedModelType(ITypeSymbol typeSymbol)
```

Comment:
it should not matter what namespace the model types are in. 
you are analyzing an assembly. all types in the assembly are candidates to be output models. a type is indeed an output model, if it's returned from a client method. The moment you decide that a type is an output model, it must have a factory in the same assembly as the model